var/spack/repos/builtin/packages/cuda/package.py

```diff
@@ -64,13 +68,22 @@ def install(self, spec, prefix):
         # https://gist.github.com/ax3l/9489132
         # for details.
 
-        runfile(
-            '--silent',         # disable interactive prompts
-            '--verbose',        # create verbose log file
-            '--override',       # override compiler version checks
-            '--toolkit',        # install CUDA Toolkit
-            '--toolkitpath=%s' % prefix
-        )
+        # CUDA 10.1+ has different cmdline options for the installer
+        if spec.satisfies('@10.1:'):
+            runfile(
+                '--silent',         # disable interactive prompts
+                '--override',       # override compiler version checks
+                '--toolkit',        # install CUDA Toolkit
+                '--installpath=%s' % prefix
+            )
+        else:
+            runfile(
+                '--silent',         # disable interactive prompts
+                '--verbose',        # create verbose log file
```

Comment:
Can you gather together the common arguments? Something like:
```python
arguments = [
    '--silent',         # disable interactive prompts
    '--override',       # override compiler version checks
    '--toolkit',        # install CUDA Toolkit
]
if spec.satisfies('@10.1:'):
    arguments.append('--installpath=%s' % prefix)
else:
    arguments.append('--verbose')
    arguments.append('--toolkitpath=%s' % prefix)

runfile(*arguments)
```

Reply:
Sure, done in the follow-up commit.

Reply:
`flake8` not happy :slightly_smiling_face: 

> var/spack/repos/builtin/packages/cuda/package.py:80: [E501] line too long (85 > 79 characters)

Reply:
Agh, fixed..

Reply:
No worries, waiting for @ax3l to give his review - if you don't hear anybody back in a couple of days ping me please!